tests/Http/Three/PostTest.php

```diff
@@ -138,6 +138,168 @@ public function testCreatingAPost()
+            'caption'          => $caption,
+            'file'             => UploadedFile::fake()->image('photo.jpg', 450, 450),
+        ]);
+
+        $response->assertStatus(200);
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'signup_id',
+                'northstar_id',
+                'media' => [
+                    'url',
+                    'original_image_url',
+                    'caption',
+                ],
+                'quantity',
+                'tags' => [],
+                'reactions' => [
+                    'reacted',
+                    'total',
+                ],
+                'status',
+                'source',
+                'remote_addr',
+                'created_at',
+                'updated_at',
+            ],
+        ]);
+
+        $this->assertDatabaseHas('posts', [
+            'signup_id' => $signup->id,
+            'northstar_id' => $signup->northstar_id,
+            'campaign_id' => $signup->campaign_id,
+            'status' => 'pending',
+            'quantity' => $quantity,
+        ]);
+
+        // Create a second post without why_participated.
+        $secondQuantity = $this->faker->numberBetween(10, 1000);
+        $secondCaption = $this->faker->sentence;
+
+        $response = $this->withAccessToken($signup->northstar_id, 'admin')->postJson('api/v3/posts', [
+            'northstar_id'     => $signup->northstar_id,
+            'campaign_id'      => $signup->campaign_id,
+            'campaign_run_id'  => $signup->campaign_run_id,
+            'quantity'         => $secondQuantity,
+            'caption'          => $secondCaption,
+            'file'             => UploadedFile::fake()->image('photo.jpg', 450, 450),
+        ]);
+
+        $response->assertStatus(200);
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'signup_id',
+                'northstar_id',
+                'media' => [
+                    'url',
+                    'original_image_url',
+                    'caption',
+                ],
+                'quantity',
+                'tags' => [],
+                'reactions' => [
+                    'reacted',
+                    'total',
+                ],
+                'status',
+                'source',
+                'remote_addr',
+                'created_at',
+                'updated_at',
+            ],
+        ]);
+
+        $this->assertDatabaseHas('posts', [
+            'signup_id' => $signup->id,
+            'northstar_id' => $signup->northstar_id,
+            'campaign_id' => $signup->campaign_id,
+            'status' => 'pending',
+            'quantity' => $quantity,
+        ]);
+    }
+
+    /**
+     * Test that a POST request to /posts without a quantity creates a new post.
+     *
+     * @return void
+     */
+    public function testCreatingAPostWithoutQuantity()
```

Comment:
could we call this `testCreatingAPostWithANullQuantity`? - sorry that's a bit nitpicky, but I think it describes the test better?

Reply:
not nitpicky - that is a lot more clear and agree! Actually updated that when adding the new post but not sure it updated in time :) Does `testCreatingAPostWithNullAsQuantity` work?

Reply:
wonderful